src/modules/commander/commander.cpp

```diff
@@ -1804,11 +1804,20 @@ Commander::run()
 			} else {
 				// use global position message to determine validity
 				const vehicle_global_position_s&global_position = _global_position_sub.get();
-				check_posvel_validity(true, global_position.eph, _eph_threshold.get(), global_position.timestamp, &_last_gpos_fail_time_us, &_gpos_probation_time_us, &status_flags.condition_global_position_valid, &status_changed);
 
 				// use local position message to determine validity
 				const vehicle_local_position_s &local_position = _local_position_sub.get();
-				check_posvel_validity(local_position.xy_valid, local_position.eph, _eph_threshold.get(), local_position.timestamp, &_last_lpos_fail_time_us, &_lpos_probation_time_us, &status_flags.condition_local_position_valid, &status_changed);
+
+				// When we are in a operator demanded position control mode and are solely reliant on optical flow, do not check position error becasue it will gradually increase throughout flight and the operator will compensate for the drift
+				bool reliant_on_opt_flow = ((estimator_status.control_mode_flags & (1<<3)) // using optical flow
+							    && !(estimator_status.control_mode_flags & (1<<2)) // not using GPS
+							    && !(estimator_status.control_mode_flags & (1<<12))); // not using vision
```

Comment:
Is it reasonable to have `1<<3` etc. hard-coded? Shouldn't we have parameters for that?

Reply:
Do you have an example of what you mean by a parameter? I could use a #define

Reply:
Use the ones we already have here https://github.com/PX4/ecl/blob/master/EKF/common.h#L178 so that we only have to change them at one place.

Reply:
I suggest pushing a copy of those defines into the uORB message, and then using the uORB flags here.

Reply:
The other thing you could consider is directly using booleans in a message (eg estimator_status_flags). Then instead of publishing constantly at 250Hz only publish when there's a change. Memory savings from packing into a bitfield will be negligible at that point and it's easier to work with, analyze, and extend.

Reply:
Those are for the EKF2_AID_MASK parameter and are different to what is required to read the control_status message https://github.com/PX4/ecl/blob/master/EKF/common.h#L417

Reply:
I will add the definitions to the estimator_status.msg file and use those